features/FEATURE_BLE/source/generic/GenericSecurityManager.cpp

```diff
@@ -853,16 +853,23 @@ void GenericSecurityManager::on_slave_security_request(
         return;
     }
 
-    if (authentication.get_secure_connections()
-        && _default_authentication.get_secure_connections()
-        && !cb->secure_connections_paired) {
-        requestPairing(connection);
+    bool pairing_required = false;
+
+    if (authentication.get_secure_connections() && !cb->secure_connections_paired
```

Comment:
this is still wrong because we don't store that quality for the key (?) needs further fixes